src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -338,12 +340,12 @@ def _choose_auto_accelerator(self) -> str:
         """Choose the accelerator type (str) based on availability."""
         if XLAAccelerator.is_available():
             return "tpu"
-        if _LIGHTNING_GRAPHCORE_AVAILABLE:
+        if _LIGHTNING_GRAPHCORE_AVAILABLE and _try_import_graphcore():
```

Comment:
You could modify the definition of `_LIGHTNING_GRAPHCORE_AVAILABLE` instead of all its usages